aws_codeseeder/resources/npm_mirror_support.py

```diff
@@ -29,7 +29,10 @@ def get_secret(secret_name: str) -> Dict[str, Dict[str, str]]:
 
 
 def main(url: str) -> None:
-    secret_name = os.environ.get("AWS_CODESEEDER_MIRROR_SECRET")
+    secret_name = os.environ.get("AWS_CODESEEDER_NPM_MIRROR_SECRET", "NO_SECRET")
+    # Backwards Compatibility
+    if secret_name == "NO_SECRET":
+        secret_name = os.environ.get("AWS_CODESEEDER_MIRROR_SECRET", "NO_SECRET")
     if secret_name:
```

Comment:
If the value is `"NO_SECRET"`, won't this be `True`? Shouldn't you check for `if secret_name != "NO_SECRET"`?

Reply:
Yea the intention is to first check if "AWS_CODESEEDER_NPM_MIRROR_SECRET" is present.

If this is false `secret_name` will be set to "NO_SECRET".

If `secret_name` is "NO_SECRET" this will then check the legacy environment variable "AWS_CODESEEDER_MIRROR_SECRET".

Is my thinking off?

Reply:
Precedence should be 
1. `AWS_CODESEEDER_NPM_MIRROR_SECRET`
2. `AWS_CODESEEDER_MIRROR_SECRET`

Reply:
I'm talking only about the evaluation of the line:
```python
if sercet_name:
```

Unless I'm misunderstanding, this will always evaluate to true. The line checks whether `secret_name` is "truthy", meaning whether it's a non-empty non-null string. If the mirror secret is not passed, `secret_name` will simply be a string with the value `"NO_SECRET"`, which will evaluate to true.

Reply:
The pattern I've used in the past for this is:

```python
secret_name = os.environ.get("AWS_CODESEEDER_NPM_MIRROR_SECRET", os.environ.get("AWS_CODESEEDER_MIRROR_SECRET"))
```

This should fallback properly and end up as `None` if neither is set.

Reply:
Ahh thanks. Github Highlights confused me. Fixed